lib/spack/spack/cmd/env.py

```diff
@@ -446,6 +448,40 @@ def env_deactivate(args):
     sys.stdout.write(cmds)
 
 
+#
+# env track
+#
+def env_track_setup_parser(subparser):
+    """track an environment from a directory in Spack"""
+    subparser.add_argument("-n", "--name", help="custom environment name")
+    subparser.add_argument("dir", help="path to environment")
+    arguments.add_common_arguments(subparser, ["yes_to_all"])
+
+
+def env_track(args):
+    src_path = os.path.abspath(args.dir)
+    if not ev.is_env_dir(src_path):
+        msg = f"cannot track environment {src_path} doesn't contain an environment"
+        raise ev.SpackEnvironmentError(msg)
+
+    if args.name:
+        name = args.name
+    else:
+        name = os.path.basename(src_path)
+
+    dst_path = ev.environment_dir_from_name(name, exists_ok=False)
```

Comment:
If this fails with a `ValueError`, Spack should print something smart about the directory name being invalid, and suggest using `spack env track --name NAME`.

Reply:
Added this in the previous commit. This workflow now exits with,

```
> spack env track path/to/test
==> error: An environment named test already exists. Set a name with,

        spack env track --name NAME path/to/test
```

Reply:
can you make this use `with:` instead of `with,`?